pkg/middleend/qos.go

```diff
@@ -89,6 +106,33 @@ func (s *Server) DeleteQosVolume(_ context.Context, in *pb.DeleteQosVolumeReques
 	return &emptypb.Empty{}, nil
 }
 
+// ListQosVolumes lists QoS volumes
+func (s *Server) ListQosVolumes(_ context.Context, in *pb.ListQosVolumesRequest) (*pb.ListQosVolumesResponse, error) {
+	log.Printf("ListQosVolume: Received from client: %v", in)
+
+	size, offset, err := server.ExtractPagination(in.PageSize, in.PageToken, s.Pagination)
+	if err != nil {
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+
+	volumes := []*pb.QosVolume{}
+	for _, qosVolume := range s.volumes.qosVolumes {
+		volumes = append(volumes, proto.Clone(qosVolume).(*pb.QosVolume))
+	}
+	sort.Sort(qosVolumesSortedByID(volumes))
```

Comment:
why you need sort ?

Reply:
I'd keep sort since map ordering is [unpredictable](https://stackoverflow.com/questions/55925822/why-are-iterations-over-maps-random) in Go even if we run a range loop over the same map multiple times.